torch/_inductor/kernel/bmm.py

```diff
@@ -87,6 +89,24 @@ def bmm_grid(b, m, n, meta):
 
 @register_lowering(aten.bmm)
 def tuned_bmm(mat1, mat2, *, layout=None):
+    # Make the inputs of bmm contiguous
+    # because bmm cpu implementation does contiguous() if not
+    # this is to avoid additional copies in bmm
+    def do_bmm_input_contiguous(t, meta_t):
+        sizes = meta_t.meta["val"].size()
+        strides = meta_t.meta["val"].stride()
+        if not ir.is_contiguous_or_transposed(sizes, strides):
+            t = ir.ExternKernel.require_contiguous(t)
+        return t
+
+    if all(x.get_device().type == "cpu" for x in [mat1, mat2]):
+        if not ir.is_storage_and_layout(mat1):
```

Comment:
How about `mat1` or `mat2` has flexible layout? Shall we also apply the similar logic here?

Reply:
Thanks and modified!